projects/plugins/jetpack/extensions/blocks/tiled-gallery/utils/index.js

```diff
@@ -15,31 +15,36 @@ export function isSquareishLayout( layout ) {
 	return [ 'circle', 'square' ].includes( layout );
 }
 
+const SERVER_OBJECT_NAME = 'JetpackWoAData';
+
 /**
  * Build src and srcSet properties which can be used on an <img />
  *
- * @param {Object} img        Image
- * @param {number} img.height Image height
- * @param {string} img.url    Image URL
- * @param {number} img.width  Image width
+ * @param {Object} img -        Image
+ * @param {number} img.height - Image height
+ * @param {string} img.url -    Image URL
+ * @param {number} img.width -  Image width
  *
- * @param {Object} galleryAtts Gallery attributes relevant for image optimization.
- * @param {string} galleryAtts.layoutStyle Gallery layout. 'rectangular', 'circle', etc.
- * @param {number} galleryAtts.columns     Gallery columns. Not applicable for all layouts.
+ * @param {Object} galleryAtts - Gallery attributes relevant for image optimization.
+ * @param {string} galleryAtts.layoutStyle - Gallery layout. 'rectangular', 'circle', etc.
+ * @param {number} galleryAtts.columns -     Gallery columns. Not applicable for all layouts.
  *
- * @return {Object} Returns an object. If possible, the object will include `src` and `srcSet`
+ * @returns {Object} Returns an object. If possible, the object will include `src` and `srcSet`
  *                  properties {string} for use on an image.
  */
 export function photonizedImgProps( img, galleryAtts = {} ) {
 	if ( ! img.height || ! img.url || ! img.width ) {
 		return {};
 	}
 
-	// Do not Photonize images that are still uploading or from localhost
+	// Do not Photonize images that are still uploading, are from localhost, or are private + atomic
+	const isPrivateAtomic =
+		window[ SERVER_OBJECT_NAME ]?.isPrivateSite && window[ SERVER_OBJECT_NAME ]?.isAtomicSite;
```

Comment:
It feels a little odd to not have a size maximum check here, but that should really be taken care of by the `upload_max_size` value.

Reply:
Instead of pulling the data that way, could we use `getJetpackData` instead? Here is an example of it in use:
https://github.com/Automattic/jetpack/blob/e13487d0dbbe4c743739b6f1a3e37c143a8c2614/projects/plugins/jetpack/extensions/shared/is-active.js#L1-L18